module/VuFindApi/src/VuFindApi/Controller/SearchApiController.php

```diff
@@ -304,89 +336,161 @@ public function searchAction()
         $request = $this->getRequest()->getQuery()->toArray()
             + $this->getRequest()->getPost()->toArray();
 
+        $isCursorSearch = ($request['resumptionToken'] ?? false);
+        try {
+            if ($isCursorSearch) {
+                $response = $this->doCursorSearch($request);
+            } else {
+                $response = $this->doDefaultSearch($request);
+            }
+        } catch (Exception $e) {
+            return $this->output([], self::STATUS_ERROR, 400, $e->getMessage());
```

Comment:
Since this is a very broad exception catch, is there any danger of leaking sensitive information here if something unexpected goes wrong? Would it be better to catch more specific exceptions for messages we want to pass through, and to use a more generic message (perhaps accompanied by debug logging) for unexpected exceptions?

Reply:
(This also seems to be a pre-existing issue... maybe too big a problem to tackle in this PR, but at least something to consider for the future).